packages/core/register/src/register.js

```diff
@@ -0,0 +1,58 @@
+import Parcel from '@parcel/core';
+import {addHook} from 'pirates';
+import process from 'process';
+import path from 'path';
+import syncPromise from '@parcel/utils/lib/syncPromise';
+
+let revert = null;
+const DEFAULT_CLI_OPTS = {
+  watch: false
+};
+
+// The filehandler returns a function that transforms the code
+function fileHandler(opts) {
+  let parcel = new Parcel({
+    entries: [path.join(process.cwd(), 'index.js')],
+    cliOpts: opts
+  });
+
+  return function(code, filename) {
+    // Not sure if this should work?
+    let result = syncPromise(
+      parcel.transform(
```

Comment:
You could use `parcel.runTransform` here to just run the transformer and not try to update the graph or watch any files.

Reply:
Awesome, that's probably the function I was looking for :)

However I'm having an issue with runtransform as the pipeline always ends up having `undefined` as values. I'll try to debug and fix this asap.